docs/datastorage/datafixers.md

```diff
@@ -0,0 +1,107 @@
+The DataFixer System
+====================
+
+The DataFixer system is provided by vanilla, and allows for fine-grained migrations and adjustments to the world format. Forge expands this system to allow mods to register and use their own fixers.
+
+What would I use the DataFixer for?
+-----------------------------------
+Here are some things vanilla uses datafixers for:
+
+* Renaming the old `HealF` health tag to `Health` (1.9)
+* Replacing all integer ID's in ItemStacks with registry names (~ 1.9)
+* Renaming the ID's of all entities and tile entities from UpperCamel names like `Blaze` to proper registry names like `minecraft:blaze` (1.11)
+* Splitting entities that were variants of each other to separate ID's, e.g. `Elder` tag on `minecraft:guardian` -> `minecraft:elder_guardian` (1.11)
+* Flattening all meta-variants of an Item into separate ID's, e.g. `minecraft:wool` meta 0 -> `minecraft:white_wool` (1.13)
+
+As you can see, this can be used in a very powerful manner to implement any NBT save format fixes you would like to make, without resorting to hacks such as checking for the old tag every tick and replacing it with a new tag.
+
+Terminology and Interfaces
+--------------------------
+
+### IFixType
+This interface represents keys to identify unique situations where NBT tags should be fixed. Its primary implementation is the enum `FixTypes`.
+Example values: `ITEM_INSTANCE` (an individual ItemStack's NBT tag), `CHUNK` (the entire NBT tag of a Chunk), `BLOCK_ENTITY` (the entire NBT tag of a Tile Entity)
+
+### DataFixer
+This is the main entry point for invoking the fixer system, and it also manages registration of fixes and walkers.
+
+### IFixableData
+This interface represents a single "fix" you would like to make, such as one of the examples listed above.
+Fixes are registered to and only run for one `IFixType`.
+
+The interface contains two methods:
+	* `int getVersion()`: This method returns the current save format version of your mod. When the system sees data that has an older version than the result of this method, it will run the fixer.
+	* `NBTTagCompound fixTagCompound(NBTTagCompound compound)`: Accepts the full NBT tag of the requested `IFixType` (i.e. the entire ItemStack, Chunk, or TileEntity), returns the fixed version
```

Comment:
"Entire ItemStack" -> Be more specific -> Do you want the nbt tag as returned by `ItemStack#serializeNBT` or the nbt tag contained in `tag` in the root tag?